test/toolbox_image.sh

```diff
@@ -12,243 +12,65 @@ ARM_EMU_BIN="${ARM_EMU_BIN:-}"
 
 SYSTEM_UPDATE_DIR="/etc/system_update"
 SYSTEM_EXECUTABLE_DIR="/sbin"
-SYSTEM_UPDATE_ENTRYPOINT="start_update.sh"
 PREPARE_DISK_COMMAND="${SYSTEM_UPDATE_DIR}.d/30_prepare_disk.sh"
-JEDI_PARTITION_TABLE_FILE_NAME="${SYSTEM_UPDATE_DIR}/jedi_emmc_sfdisk.table"
 UPDATE_FILES_COMMAND="${SYSTEM_UPDATE_DIR}.d/50_update_files.sh"
-START_UPDATE_COMMAND="${SYSTEM_EXECUTABLE_DIR}/${SYSTEM_UPDATE_ENTRYPOINT}"
-
-# Test storage device parameters.
-# All storage start and and parameters are sector numbers. actual byte positions
-# are calculated by multiplying the sector number with the sector size.
-STORAGE_DEVICE_IMG="/tmp/storage_device.img"
-BYTES_PER_SECTOR="512"
-MIN_PARTITION_SIZE="78124"    # 40 MiB (enough for f2fs and ext4)
-STORAGE_DEVICE_SIZE="7553024" # sectors, about 3.6 GiB
-#BOOT_START="2048"             # offset 1 MiB
-ROOTFS_START="67584"          # offset 33 MiB
-USERDATA_START="1998848"      # offset 976 MiB
-LOOP_STORAGE_DEVICE=""
-
-PARTITION_TABLE_FILE_NAME="partition_table"
-PARTITION_TABLE_FILE="${SYSTEM_UPDATE_DIR}/${PARTITION_TABLE_FILE_NAME}"
+START_UPDATE_COMMAND="${SYSTEM_EXECUTABLE_DIR}/start_update.sh"
+JEDI_PARTITION_TABLE_FILE="${SYSTEM_UPDATE_DIR}/jedi_emmc_sfdisk.table"
+JEDI_EXCLUDE_LIST_FILE="${SYSTEM_UPDATE_DIR}/jedi_update_exclude_list.txt"
 TMP_TEST_IMAGE_FILE="/tmp/test_file.img"
 
-UPDATE_ROOTFS_SOURCE="/tmp/update_source"
-UPDATE_EXCLUDE_LIST_FILE="config/jedi_update_exclude_list.txt"
-TEST_UPDATE_ROOTFS_FILE="test/test_rootfs.tar.xz"
-TEMP_TEST_UPDATE_ROOTFS_FILE="rootfs-v1.2.3.tar.xz"
+TEST_OUTPUT_FILE="$(mktemp -d)/test_results_$(basename "${0%.sh}").txt"
 
-TEST_OUTPUT_FILE="$(mktemp -d)/test_results.txt"
+toolbox_image=""
+toolbox_root_dir=""
 
-overlayfs_dir=""
-rootfs_dir=""
-update_mount=""
-
-result=0
-
-is_dev_setup_mounted=false
 exit_on_failure=false
+result=0
 
+# shellcheck source=test/include/armhf_emulation.sh
+. "test/include/armhf_emulation.sh"
```

Comment:
while . is a well known an fully acceptable abbreviation of 'source', i'd recommend to use the word source here instead for two reasons. It eases reading and matches the 'source=' statement above.

Finally, i'd recommend putting includes at the very top of the file. there should not be any dependencies which will 'force' that by having it on top.